### ISSUE TITLE
Blazor server-side project with windows authentication does not work from command line

### ISSUE THREAD

**Issue by akorchev** [dotnet/aspnetcore#23145]
### Describe the bug
If you create a blazor server-side application with `dotnet new blazorserver --auth Windows` it doesn't display the currently logged user if run with `dotnet run`. 

### To Reproduce

1. Create a new application by running  `dotnet new blazorserver --auth Windows`
2. Run the application with `dotnet run`
3. The LoginDisplay component does not show .

I happen to know why this happens :) The template works with Visual Studio Professional because it can run it via IISExpress. The project template creates an IISExpress configuration which works as expected. However running from VS Code or command line does not do the windows authentication.

I see a few ways to handle this better:

- Explain either in the output of `dotnet new` or in documentation that this template only works with IISExpress.
- Use  [Microsoft.AspNetCore.Authentication.Negotiate](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth?view=aspnetcore-3.0&tabs=visual-studio#kestrel) in the `dotnet new` template. It works with `dotnet run`.

**Comment by javiercn** [comment]
@akorchev thanks for contacting us.

We'll update the docs to mention both posibilities. @guardrex can you add this to wherever we talk about Windows auth?

**Comment by javiercn** [comment]
@Tratcher is it correct/safe to reference Microsoft.AspNetCore.Authentication.Negotiate even if you plan to run on IIS? If that's the case we could make that change.

**Comment by akorchev** [comment]
@javiercn there is at least one difference when using `Microsoft.AspNetCore.Authentication.Negotiate` instead of IISExpress - the user isn't automatically logged in with the former.

**Comment by guardrex** [comment]
Our general coverage is here ...

https://docs.microsoft.com/en-us/aspnet/core/security/authentication/windowsauth?view=aspnetcore-3.1&tabs=visual-studio#kestrel

Our pattern is generally not to repeat our ASP.NET Core topic coverage for Blazor Server apps. We say ...

> Blazor Server apps are configured for security in the same manner as ASP.NET Core apps. For more information, see the articles under Overview of ASP.NET Core Security.

https://docs.microsoft.com/en-us/aspnet/core/security/blazor/server/?view=aspnetcore-3.1&tabs=visual-studio#secure-an-existing-app

... but I don't like the heading "Secure an existing app" above that line. It applies to both new and existing apps. I'll fix that right now.

**Comment by akorchev** [comment]
@guardrex I personally know the solution but wanted to prevent other developers from scratching their heads. The documentation for `dotnet new blazorserverside --auth Windows` should at least say "Works with IISExpress only".

**Comment by guardrex** [comment]
Sounds more like a more general launch profile issue at this point. We document ...

> Profiles can be selected from Visual Studio **_or using `dotnet run --launch-profile <Profile Name>`._** (emphasis added)

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-3.1#development-and-launchsettingsjson

I haven't tried to run a `blazorserver` template-based app with `dotnet run` and specifying the IIS Express profile. Actually, I haven't tried it for any ASP.NET Core app in a long, long time. Today, trying it with the following command ...

```dotnetcli
dotnet run --launch-profile "IIS Express"
```

... results in ...

```
The launch profile "IIS Express" could not be applied.
The launch profile type 'IISExpress' is not supported.
```

... for any app. Blazor or not.

Clearly that's misleading because we know these apps can run from VS with IIS Express.

I suspect that once this is worked out that an app using WinAuth and that's run from the CLI will work.

I'll keep an :ear: open here for @Tratcher's advice.

**Comment by akorchev** [comment]
@guardrex IISExpress is Windows-only though. Probably the Windows authentication template shouldn't be available in other OS. Or it should use Microsoft.AspNetCore.Authentication.Negotiate which is supposed to work cross platform.

**Comment by guardrex** [comment]
I think devs should be able to create such an app on ... say ... macOS ... and have it run on Windows on IIS Express just by moving it.

The Kestrel scenario is well covered in the docs, so I think that's a product unit issue at this point if they want to do something else with the template, like a switch that can set that up ✨ *auto-magically* ✨.

There's a broad challenge in that the framework is xplat but not *all* of the tooling and scenarios work xplat. **Feature Request!** ... *Can we have IIS/IIS Express xplat?!* 😁 lol .... Yikes! I just got an evil :eye: from the :pray: *server gods* 🙏 joking about that. lol

Seriously tho, I think if at least the profile run for IIS Express is clarified that WinAuth will work, and that's something that can be improved in the *Environments* topic and possibly the *WinAuth* and *Blazor Server security* topics. I'll work it on the https://github.com/dotnet/AspNetCore.Docs/issues/18859 issue that I opened.

**Comment by Tratcher** [comment]
The Negotiate package automatically defers to the server if the server supports windows auth, so yes you could add it to the template. As for logging in by default, I think you need to set the fallback policy for that.

**Comment by javiercn** [comment]
@Tratcher thanks for the details.

Moving this to next sprint planning to consider updating the template to include the necessary changes for windows auth. Those changes would be to add the mentioned package to the csproj file, and to configure the appropriate authpolicy to ensure the user is authenticated.

**Comment by javiercn** [comment]
The docs here look complete enough, as for any template change we might want to do in this area, let's wait until 5.0 changes are merged with master.

**Comment by javiercn** [comment]
@Tratcher didn't you fix this in 6.0 recently? (Since Kestrel is the new default)

**Comment by Tratcher** [comment]
Fixed in 6.0 RTM by https://github.com/dotnet/aspnetcore/pull/37607
